### DISCUSSION TITLE
Error: cannot activate the 'Python' ext. because it depends on the 'Jupyter' ext.

### DISCUSSION THREAD

**Discussion by StefanoFrazzetto** [community/community#3571] [Codespaces] [answered]
I just opened a project in VS Code from Codespaces and I'm getting the following error:
> Cannot activate the 'Python' extension because it depends on the 'Jupyter' extension, which is not loaded. Would you like to reload the window to load the extension?

This is not the first time this issue come up, and reloading the window doesn't solve anything, but rebuilding the container seems to work. Any idea how I could fix this?

Info from VS Code 'About':
```
Version: 1.54.3 (system setup)
Commit: 2b9aebd5354a3629c3aba0a5f5df49f43d6689f8
Date: 2021-03-15T10:55:45.459Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042
```

**UPDATE:**
As pointed out in the comments below, the issue seems to be fixed by **reinstalling the extension**, either the same one or a previous one, and **reloading VS Code**. Thanks everyone for sharing your solutions, too!

**Comment by 2percentsilk** [comment]
Hey @StefanoFrazzetto,

Is the repo you're using public / could you share that here? Are you using a repo with a `.devcontainer` folder?

**Reply by StefanoFrazzetto** [reply]
Hey @2percentsilk,

The repo I used the first time is private, but I created a [new one](https://github.com/StefanoFrazzetto/CodespacesFeedback) and I have the same issue

![codespaces_feedback_cannot_activate_python](https://user-images.githubusercontent.com/9978588/112547970-664e4500-8db3-11eb-85fa-eac57537d92d.png)

I did some digging and I have I found the issue:

From _Output > Log (Remote Server)_
> [2021-03-25 21:24:20.255] [remoteagent] [info] Extension host agent started.
[2021-03-25 21:24:20.394] [remoteagent] [info] Deleted from disk littlefoxteam.vscode-python-test-adapter /home/codespace/.vscode-remote/extensions/littlefoxteam.vscode-python-test-adapter-0.6.7
[2021-03-25 21:24:21.410] [remoteagent] [info] [::ffff:127.0.0.1][ee4635bc][ManagementConnection] New connection established.
[2021-03-25 21:24:21.476] [remoteagent] [info] [::ffff:127.0.0.1][d6566826][ExtensionHostConnection] New connection established.
[2021-03-25 21:24:21.485] [remoteagent] [info] [::ffff:127.0.0.1][d6566826][ExtensionHostConnection] <1747> Launched Extension Host Process.
**[2021-03-25 21:24:21.827] [remoteagent] [error] /home/codespace/.vscode-remote/extensions/ms-toolsai.jupyter-2021.5.685944597 Extension is not compatible with Code 1.54.3. Extension requires: 1.55.0-insider.**

I tried installing VS Code Insiders and, as expected, the extension is now working fine
> Version: 1.55.0-insider (user setup)
Commit: 6ebe2a14f3722bc2582ecd8108d0a71dfdd46036
Date: 2021-03-25T16:25:06.257Z
Electron: 11.3.0
Chrome: 87.0.4280.141
Node.js: 12.18.3
V8: 8.7.220.31-electron.0
OS: Windows_NT x64 10.0.19042

Downgrading to `Jupyter v2021.3.619093157` in `VS Code v1.54.3` and reloading the window seemed to have fixed the issue, at least apparently, but then I was asked to select a local Python interpreter, so I couldn't run any code. I then re-upgraded to `Jupyter v2021.3.684299474`, closed the window, triggered VS Code to open again (from the Codespaces webpage), and it started working normally.

While I can't explain what happened with the last part, I hope the info here can be useful.
Let me know if there's anything else I can do!

**Comment by thistlillo** [comment]
In my case, all of a sudden CTRL+P for executing a command stopped working: I could select the command, but "enter" did not execute the command. After relaunching VSC, I got this error too. Using the latest versions of everything on Windows 10 Pro.

**Reply by luxtsii** [reply]
Having the same issue, seemingly at the exact same time as you. Hopefully someone has a solution.

**Reply by ioch96** [reply]
same issue

**Reply by ftrbrum** [reply]
I am having it too.

**Reply by gablabc** [reply]
Me too, it just stopped working.

**Reply by staneta** [reply]
Same

**Reply by yiidokun** [reply]
I had the same issue. There were no "enable" Jupyter in extensions. I think the latest VS Code update caused it. I reinstalled the latest version of the Jupyter extension from "Install Another Version" option worked for me. Worth giving it a try.

**Reply by rahulp7801** [reply]
It has been fixed for me just now

**Reply by luxtsii** [reply]
> I had the same issue. There were no "enable" Jupyter in extensions. I think the latest VS Code update caused it. I reinstalled the latest version of the Jupyter extension from "Install Another Version" option worked for me. Worth giving it a try.

Legendary fix, thank you very much.

**Reply by lazyprogrammerrr** [reply]
same issue.

**Reply by ftrbrum** [reply]
it just started working for me again

**Reply by staneta** [reply]
ctrl+shift+p -> reinstall extension -> jupyter. fixes the problem.

**Reply by TahaM1** [reply]
> ctrl+shift+p -> reinstall extension -> jupyter. fixes the problem.

Thank you.

**Reply by jaydenlsf** [reply]
extensions -> search for jupyter -> arrow key next to 'uninstall' -> install another version -> select the second latest version

Reload your vscode and it should be fixed.

**Reply by younesysf** [reply]
> ctrl+shift+p -> reinstall extension -> jupyter. fixes the problem.

worked for me. thanks

**Reply by slhck** [reply]
For me this worked:

- Uninstall the Python extension
- Uninstall the Jupyter extension
- Uninstall the Pylance extension
- Install the Python extension (it installs Pylance and Jupyter)
- Reload VS Code

**Comment by EmilyJYN** [comment, marked as answer]
Hey, I just had the same error and have fixed it. I simply uninstalled Jupyter and python extensions for the remote address, re-installed them again, and then refresh VSCode. Everything is working now.

**Reply by thistlillo** [reply]
At the end I reinstalled jupyter and could work again. However, in the last days several extensions have broken (e.g. git lens) and I have not been able to fix them by simply re-installing.
